### DISCUSSION TITLE
How to prevent Vieb to show help in new instance (**Private Mode**)

### DISCUSSION THREAD

**Discussion by poslat-pismo** [Jelmerro/Vieb#510] [Show and tell]
Hello, I launch **Vieb** in private mode using this script:
`#!/bin/sh

TMPDIR=/tmp/priv-$$

mkdir "${TMPDIR}"
echo '{"tabs"[{"container":"temp1","muted":false,"url":"vieb://newtab"}]}' > "${TMPDIR}/tabs"
echo "quit" > "${TMPDIR}/commandhist"
cat << EOF > "${TMPDIR}/viebrc"
set containershowname=never
set containerkeeponreopen=false
set containercolors="temp\d+~lightblue"
set containernewtab=temp%n
set containerstartuppage=""
EOF
vieb --datafolder="${TMPDIR}"
rm -rf "${TMPDIR}"
`

**Comment by Jelmerro** [comment]
Your answer is indeed correct, since then there is already a page visited, instead of it being a new instance, you will not get the help page on startup.

Alternatively, you can follow the described steps at `:help storenewvisits` and related settings, to disable all the settings responsible for store history/localstorage etc. to basically disable all storage and thus no longer require the removal of the entire folder.
For example, with these options set:
![image](https://github.com/Jelmerro/Vieb/assets/1696674/bc9934cf-9796-40bb-9fdd-8d9e7e9f5c1a)
The files that remain in a datafolder after closing will look similar to this:
![image](https://github.com/Jelmerro/Vieb/assets/1696674/22627fb2-ce27-4be5-abb8-04c910fbf237)
To al least store the history temporarily while it is running, you could remove the option that disables `storenewvisits`, while you can also get back the option to restore tabs while running by removing the `nokeeprecentlyclosed` from these settings.

**Reply by poslat-pismo** [reply]
Still all URLs are in human readable 'tabs' file ((
so I did:

`
if [ ! -d "${HOME}/.config/priv" ]; then

echo New session

mkdir -p "${HOME}/.config/priv"

echo "quit" > "${HOME}/.config/priv/commandhist"

echo '{"closed":[],"id":0,"pinned":[],"tabs":
[{"container":"temp","muted":false,"url":"vieb://newtab"}]}' > 

"${HOME}/.config/priv/tabs"

cat << EOF > "${HOME}/.config/priv/viebrc"

set containershowname=never

set containerkeeponreopen=false

set containercolors="temp\d+~lightblue"

set containernewtab=temp

set clearhistoryinterval=session

set clearcookiesonquit

set cleardownloadsonquit

set clearlocalstorageonquit

set explorehist=none

set commandhist=none

set favicons=nocache

EOF

fi

vieb --datafolder="${HOME}/.config/priv"
echo '{"closed":[],"id":0,"pinned":[],"tabs":[{"container":"temp","muted":false,"url":"vieb://newtab

"}]}' > "${HOME}/.config/priv/tabs"

`
